addons/hr_holidays/models/resource.py

```diff
@@ -1,10 +1,136 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from pytz import timezone, UTC
+from odoo import fields, models, api, _
+from datetime import timedelta
+from odoo.tools import date_utils
+from odoo.osv import expression
 
 
 class CalendarLeaves(models.Model):
     _inherit = "resource.calendar.leaves"
 
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
+
+    def _get_domain(self, time_domain_dict):
+        domain = []
+        for record in time_domain_dict:
+            domain = expression.OR([domain, [
+                    ('date_to', '>', record['date_from']),
+                    ('date_from', '<', record['date_to'])]
+            ])
+        return expression.AND([domain, [('state', '!=', 'refuse'), ('active', '=', 'True')]])
+
+    def _get_time_domain_dict(self, time_domain_dict=None):
+        if not time_domain_dict:
+            time_domain_dict = []
+        for record in self:
+            if not record['resource_id']:
+                time_domain_dict.append({
+                    'date_from' : record['date_from'],
+                    'date_to' : record['date_to']
+                })
+        return time_domain_dict
```

Comment:
Following the comment in `write` + list interpretation, also it is weird to access record values using ['attr'] syntax:
```suggestion
    def _get_time_domain_dict(self):
        return [{
            'date_from': record.date_from,
            'date_to': record.date_to,
        } for record in self if record['resource_id']]
```